osu.Game/Online/Leaderboards/LeaderboardScore.cs

```diff
@@ -58,6 +58,9 @@ public class LeaderboardScore : OsuClickableContainer, IHasContextMenu
         [Resolved(CanBeNull = true)]
         private DialogOverlay dialogOverlay { get; set; }
 
+        [Resolved(CanBeNull = true)]
+        private SongSelect songSelect { get; set; }
```

Comment:
I'm very not much a fan of mod changes being done through `SongSelect` - it necessitates a lot of introductions of the `immediate` flag and the exposition of `ModSelect`. I would instead propose to inject

```csharp
[Resolved(CanBeNull = true)]
private Bindable<IReadOnlyList<Mod>> selectedMods { get; set; }
```

directly from `OsuGame` which seems to work as well. Not sure whether there are any particular reasons against doing that, though.

Reply:
Actually I don't need to expose `ModSelect` and do any of the deselection stuff. `songSelect.Mods.Value = score.Mods` already does replace the old mods.

Reply:
that works too, I suppose.